sdk/schema/executable.go

```diff
@@ -37,6 +37,29 @@ type CredentialUsage struct {
 	// set in the credential schema, so should only be used if this executable requires a custom configuration, that deviates
 	// from the way the credential is usually provisioned.
 	Provisioner sdk.Provisioner
+
+	// (Optional) What this credential will be used for by the executable.
+	Description string
+
+	// (Optional) Instead of requiring a specific credential, have the user select from a list of compatible credentials.
+	// Mutually exclusive with: `Name` and `Plugin`.
+	SelectFrom *CredentialSelection
+
+	// (Optional) Whether the exectuable needs authentication for this credential. Works side by side with the executable's
+	// `NeedsAuth`, which can still be used for more generic authentications opt-outs, such as the help flag.
+	NeedsAuth sdk.NeedsAuthentication
+
+	// Whether this credential is needed for the executable to run. If set to true, the executable cannot run without provisioning this credential.
+	Optional bool
+}
+
+type CredentialSelection struct {
+	// ID helps identify credentials chosen in this selection. This must be unique in relation to other selections specified in usages within its executable.
+	ID string
+	// IncludeAllCredentials specifies whether all credentials defined in all plugins should be included in the selection prompt when configuring this credential use.
```

Comment:
We can shorten this, given the context that's already there:
```suggestion
	// IncludeAllCredentials specifies whether this selection should contain all credentials defined in all plugins.
```

Reply:
Done, thanks!